File: From_Last_Year/Jutge/Week 4/Read_mapping_1.cpp
#include <iostream>
#include <string>
#include <map>
/*
READ_MAPPING1(S)
	let de b empty dictionary
	for i = 0 to |S| do
		d[S[i:]] <-  i+1
		for x in sorted(d) do
			print(d[x])
*/
using namespace std;

int main(){

    map <string, int> d;

    string seq;

    cin >> seq;

    for(int i = 0; i < seq.length(); i++){
        d[seq.substr(i,seq.length())] = i+1;
    }

    for (const auto &x : d){
        cout << x.second << endl;
    }
    
}

File: From_Last_Year/Jutge/Week 4/Read_mapping_1.py
from jutge import read
import sys
"""
READ_MAPPING1(S)
	let d be empty dictionary
	for i = 0 to  |S| do
		d[S[i:]] <-  i+1
		for x in sorted(d) do
			print(d[x])
	
"""

seq = sys.stdin.readline().strip()

d = {}

for i in range(len(seq)):
	d[seq[i:]] = i+1
		
for x in sorted(d):
	print(d[x])
		

